opensearchpy/connection/http_async_requests.py

```diff
@@ -0,0 +1,287 @@
+                raise ImproperlyConfigured("client_key is not a path to a file")
+            if client_cert and client_key:
+                ssl_context.load_cert_chain(client_cert, client_key)
+            elif client_cert:
+                ssl_context.load_cert_chain(client_cert)
+
+        self.headers.setdefault("connection", "keep-alive")
+        self.loop = loop
+        self.session = None
+
+        # Parameters for creating an aiohttp.ClientSession later.
+        self._limit = maxsize
+        self._http_auth = http_auth
+        self._ssl_context = ssl_context
+
+    async def perform_request(
+        self, method, url, params=None, body=None, timeout=None, ignore=(), headers=None
+    ):
+        if self.session is None:
+            await self._create_aiohttp_session()
+        assert self.session is not None
+        orig_body = body
+        url_path = self.url_prefix + url
+        if params:
+            query_string = urlencode(params)
+        else:
+            query_string = ""
+
+
+        # There is a bug in aiohttp that disables the re-use
+        # of the connection in the pool when method=HEAD.
+        # See: aio-libs/aiohttp#1769
+        is_head = False
+        if method == "HEAD":
+            method = "GET"
+            is_head = True
+
+        # Top-tier tip-toeing happening here. Basically
+        # because Pip's old resolver is bad and wipes out
+        # strict pins in favor of non-strict pins of extras
+        # our [async] extra overrides aiohttp's pin of
+        # yarl. yarl released breaking changes, aiohttp pinned
+        # defensively afterwards, but our users don't get
+        # that nice pin that aiohttp set. :( So to play around
+        # this super-defensively we try to import yarl, if we can't
+        # then we pass a string into ClientSession.request() instead.
+        url = self.url_prefix + url
+        if query_string:
+            url = "%s?%s" % (url, query_string)
+        url = self.host + url
+
+        timeout = aiohttp.ClientTimeout(
+            total=timeout if timeout is not None else self.timeout
+        )
+
+        req_headers = self.headers.copy()
+        if headers:
+            req_headers.update(headers)
+
+        if self.http_compress and body:
+            body = self._gzip_compress(body)
+            req_headers["content-encoding"] = "gzip"
+
+        req_headers = {**req_headers, **self._http_auth(method, url, query_string, body)}
+
+        start = self.loop.time()
+        try:
+
+            async with self.session.request(
+                method,
+                url,
+                data=body,
+                headers=req_headers,
+                timeout=timeout,
+                fingerprint=self.ssl_assert_fingerprint,
+            ) as response:
+                if is_head:  # We actually called 'GET' so throw away the data.
+                    await response.release()
+                    raw_data = ""
```

Comment:
Should it be `None`?

Reply:
https://github.com/harshavamsi/opensearch-py/blob/0ec369ffd5632191d1f4b1c6191bb4d381f2f9c2/opensearchpy/_async/http_aiohttp.py#L290 We're inheriting the upstream class, looks like it's instantiated the same way there.